students/kulon_aleksander/lesson_04/test_length_of_the_segment.py

```diff
@@ -0,0 +1,29 @@
+import math
+import pytest
+from lesson_03.length_of_the_segment import distance
+
+
+class TestDistanceFunction(object):
+
+    def test_input_none(self):
+        with pytest.raises(TypeError):
+            distance(None, 1, 2, 3)
+
+    def test_input_string(self):
+        with pytest.raises(TypeError):
+            distance('aoeu', 1, 2, 3)
+
+    def test_zero_distance(self):
+        assert distance(1, 1, 1, 1) == 0
+
+    def test_corner_cases(self):
+        assert distance(-1, -1, -2, -2) == math.sqrt(2)
+        assert distance(0, 1, 0, 2) == 1
```

Comment:
Tip: Don't make too many assertions per test. Better is to have more tests than one big unit test that is checking few things. A test should only be able to fail for a single reason, namely that the thing it tests doesn’t behave as expected.